File: frontend/src/api/environment.ts
import request from './request'

export interface EnvironmentItem {
  id: number
  name: string
  env_type: string
  region: string | null
  base_url: string | null
  mqtt_broker_url: string | null
  db_url: string | null
  redis_url: string | null
  ai_evaluator_url: string | null
  wiremock_url: string | null
  payment_endpoint: string | null
  status: string
  config: Record<string, any> | null
  description: string | null
  created_at: string
  updated_at: string
}

export interface HealthCheckItem {
  id: number
  env_id: number
  component: string
  status: string
  latency_ms: number | null
  details: Record<string, any> | null
  checked_at: string
}

export interface SnapshotItem {
  id: number
  env_id: number
  name: string
  snapshot_type: string
  state_data: Record<string, any> | null
  notes: string | null
  created_at: string
}

export function getEnvironments(params?: { env_type?: string; status?: string }) {
  return request.get('/environments', { params }) as Promise<EnvironmentItem[]>
}

export function getEnvironment(id: number) {
  return request.get(`/environments/${id}`) as Promise<EnvironmentItem>
}

export function createEnvironment(data: Partial<EnvironmentItem>) {
  return request.post('/environments', data) as Promise<EnvironmentItem>
}

export function updateEnvironment(id: number, data: Partial<EnvironmentItem>) {
  return request.put(`/environments/${id}`, data) as Promise<EnvironmentItem>
}

export function deleteEnvironment(id: number) {
  return request.delete(`/environments/${id}`) as Promise<{ ok: boolean }>
}

export function checkEnvironmentHealth(id: number) {
  return request.post(`/environments/${id}/health-check`) as Promise<HealthCheckItem[]>
}

export function createSnapshot(envId: number, data: { name: string; snapshot_type?: string; notes?: string }) {
  return request.post(`/environments/${envId}/snapshots`, data) as Promise<SnapshotItem>
}

export function getSnapshots(envId: number) {
  return request.get(`/environments/${envId}/snapshots`) as Promise<SnapshotItem[]>
}

export function restoreSnapshot(snapshotId: number) {
  return request.post(`/environments/snapshots/${snapshotId}/restore`) as Promise<EnvironmentItem>
}

File: backend/app/schemas/environment.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class EnvironmentBase(BaseModel):
    name: str
    env_type: str = "staging"
    region: str | None = None
    base_url: str | None = None
    mqtt_broker_url: str | None = None
    db_url: str | None = None
    redis_url: str | None = None
    ai_evaluator_url: str | None = None
    wiremock_url: str | None = None
    payment_endpoint: str | None = None
    config: dict | None = None
    description: str | None = None


class EnvironmentCreate(EnvironmentBase):
    pass


class EnvironmentUpdate(BaseModel):
    name: str | None = None
    env_type: str | None = None
    region: str | None = None
    base_url: str | None = None
    mqtt_broker_url: str | None = None
    db_url: str | None = None
    redis_url: str | None = None
    ai_evaluator_url: str | None = None
    wiremock_url: str | None = None
    payment_endpoint: str | None = None
    status: str | None = None
    config: dict | None = None
    description: str | None = None


class EnvironmentResponse(EnvironmentBase):
    model_config = ConfigDict(from_attributes=True)
    id: int
    status: str
    created_at: datetime
    updated_at: datetime


class HealthCheckResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    env_id: int
    component: str
    status: str
    latency_ms: float | None = None
    details: dict | None = None
    checked_at: datetime


class SnapshotResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    env_id: int
    name: str
    snapshot_type: str
    state_data: dict | None = None
    notes: str | None = None
    created_at: datetime


class SnapshotCreate(BaseModel):
    name: str
    snapshot_type: str = "manual"
    notes: str | None = None


class EnvironmentHealthSummary(BaseModel):
    environment: EnvironmentResponse
    health_checks: list[HealthCheckResponse]
    overall_status: str
